src/proxy/inject.js

```diff
@@ -148,21 +148,21 @@ function inject(target, currentGeneration, injectedMembers) {
             key,
             `(function REACT_HOT_LOADER_SANDBOX () {
           var _this = this; // common babel transpile
-          var _this2 = this; // common babel transpile          
+          var _this2 = this; // common babel transpile
```

Comment:
This is ☠️, I know that _this3 exists.
In ideal we should or inject `regenerate` in a constructor, to make these variables visible, or detect the error on regenerate, and ask to disable arrow function transpiling.